apps/console/src/features/api-resources/api/api-resources.ts

```diff
@@ -0,0 +1,307 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AsgardeoSPAClient, HttpClientInstance } from "@asgardeo/auth-react";
+import { IdentityAppsApiException } from "@wso2is/core/exceptions";
+import { HttpMethods } from "@wso2is/core/models";
+import { AxiosError, AxiosRequestConfig, AxiosResponse } from "axios";
+import useRequest, {
+    RequestErrorInterface,
+    RequestResultInterface
+} from "../../core/hooks/use-request";
+import { store } from "../../core/store";
+import { APIResourceInterface, APIResourcePermissionInterface, APIResourcesListInterface, 
+    UpdatedAPIResourceInterface } from "../models";
+
+/**
+ * Get an axios instance.
+ */
+const httpClient: HttpClientInstance = AsgardeoSPAClient.getInstance()
+    .httpRequest.bind(AsgardeoSPAClient.getInstance());
+
+/**
+ * Get API resources for the identifier validation.
+ * Only to be used for the identifier validation.
+ * 
+ * @param filter - filter.
+ * @returns `Promise<APIResourcesListInterface | IdentityAppsApiException>`
+ */
+export const getAPIResourcesForIdenitifierValidation = (
+    filter: string
+): Promise<APIResourcesListInterface> => {
+
+    const requestConfig: AxiosRequestConfig = {
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json"
+        },
+        method: HttpMethods.GET,
+        params: {
+            filter
+        },
+        url: `${store.getState().config.endpoints.apiResources}`
+    };
+
+    return httpClient(requestConfig)
+        .then((response: AxiosResponse) => {
+            if (response.status === 200) {
+                return Promise.resolve(response.data as APIResourcesListInterface);
+            } else {
+                throw new IdentityAppsApiException(
+                    response.data.description,
+                    null,
+                    response.status,
+                    response.request,
+                    response,
+                    response.config);
+            }
+        })
+        .catch((error: AxiosError) => {
+            throw new IdentityAppsApiException(
+                error.message,
+                error.stack,
+                error.response?.data?.code,
+                error.request,
+                error.response,
+                error.config);
+        });
+};
+
+/**
+ * Get API resources.
+ *
+ * @param after - after.
+ * @param before - before.
+ * @returns `Promise<APIResourcesListInterface>`
+ * @throws `IdentityAppsApiException`
+ */
+export const useAPIResources = <Data = APIResourcesListInterface, Error = RequestErrorInterface>(
```

Comment:
Rename to `useGetAPIResources` to qualify that this is a SWR GET hook.

```suggestion
export const useGetAPIResources = <Data = APIResourcesListInterface, Error = RequestErrorInterface>(
```

Also, move these to dedicated modules(`use-get-api-resources`) so we can use default exports.

Ref: https://github.com/wso2/identity-apps/blob/master/apps/console/src/features/branding/api/use-get-custom-text-preference-fallbacks.ts